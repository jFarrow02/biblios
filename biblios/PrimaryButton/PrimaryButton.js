"use strict";Object.defineProperty(exports,"__esModule",{value:true});exports["default"]=void 0;var _react=_interopRequireDefault(require("react"));var _propTypes=_interopRequireDefault(require("prop-types"));var _styledComponents=_interopRequireDefault(require("styled-components"));var _styles=_interopRequireDefault(require("../styles"));function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj}}var PrimaryButton=function PrimaryButton(props){var children=props.children,_onClick=props.onClick,disabled=props.disabled;var StyledButton=_styledComponents["default"].button.withConfig({displayName:"PrimaryButton__StyledButton",componentId:"sc-1n4yz04-0"})(["padding:15px 25px;border:none;background:",";color:",";cursor:",";font-size:18px;margin:10px 15px 10px 0px;"],function(props){return _styles["default"].getType(props.type)},function(props){if(props.disabled){return _styles["default"].gray08}return _styles["default"].white},function(props){return props.disabled?"normal":"pointer"});return _react["default"].createElement(StyledButton,{disabled:disabled,type:props.type,onClick:function onClick(){_onClick()}},children)};var _default=PrimaryButton;exports["default"]=_default;PrimaryButton.propTypes={children:_propTypes["default"].oneOfType([_propTypes["default"].string,_propTypes["default"].element]),disabled:_propTypes["default"].bool,onClick:_propTypes["default"].func,type:_propTypes["default"].oneOf(["normal","caution","alert","warning","disabled"])};PrimaryButton.defaultProps={children:"Button",disabled:false,onClick:null,type:"normal"};