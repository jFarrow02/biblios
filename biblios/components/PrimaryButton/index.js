"use strict";Object.defineProperty(exports,"__esModule",{value:true});exports["default"]=void 0;var _PrimaryButton=_interopRequireDefault(require("./PrimaryButton"));function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{"default":obj}}var _default=_PrimaryButton["default"];exports["default"]=_default;