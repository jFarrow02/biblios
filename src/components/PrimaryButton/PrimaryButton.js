import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import stylesConfig from '../styles';

const PrimaryButton = props =>{
    const { children, onClick, disabled} = props;
    const StyledButton = styled.button`
        padding: 15px 25px;
        border: none;
        background: ${ props=> { return stylesConfig.getType(props.type) }};
        color: ${ props => {
            if(props.disabled){
                return stylesConfig.gray08;
            }
            return stylesConfig.white;
        }};
        cursor: ${props => props.disabled ? 'normal' : 'pointer'};
        font-size: 18px;
        margin: 10px 15px 10px 0px;
    `;
    return(
        <StyledButton
            disabled={disabled}
            type={props.type}
            onClick={()=>{onClick()}}
        >
            {children}
        </StyledButton>
    );
}

export default PrimaryButton;

PrimaryButton.propTypes = {
    children: PropTypes.oneOfType(
        [
            PropTypes.string,
            PropTypes.element,
        ],
    ),
    disabled: PropTypes.bool,
    onClick: PropTypes.func,
    type: PropTypes.oneOf(
        [
            'normal',
            'caution',
            'alert',
            'warning',
            'disabled',
        ]
    ),
};

PrimaryButton.defaultProps = {
    children: 'Button',
    disabled: false,
    onClick: null,
    type: 'normal',
};