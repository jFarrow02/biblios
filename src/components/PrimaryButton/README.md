# PrimaryButton

**PrimaryButton** is a React functional component. It renders a button element.

## Props

|Name       |Type                   |Default        |Description                                        |
|-----------|-----------------------|---------------|---------------------------------------------------|
|children   |`{JSXElement/String}`  |'Button'       |Child elements to be rendered inside of `<button>` |
|onClick    |`{Function}`           |null           |Handler function for button's 'click' event        |