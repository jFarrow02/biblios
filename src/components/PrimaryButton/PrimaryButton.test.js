import React from 'react';
import PrimaryButton from './PrimaryButton';
import renderer from 'react-test-renderer';

it('should match a snapshot', ()=> {
    const component = renderer.create(<PrimaryButton/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});