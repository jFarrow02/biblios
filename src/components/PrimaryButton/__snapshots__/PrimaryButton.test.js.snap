// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<button
  className="PrimaryButton__StyledButton-sc-1n4yz04-0 VKVYj"
  disabled={false}
  onClick={[Function]}
  type="normal"
>
  Button
</button>
`;
